From 49f934d20b57b57598cdae55311c88cb6f0b814f Mon Sep 17 00:00:00 2001
From: Leo Hsu <leohsu@google.com>
Date: Wed, 6 Mar 2019 15:25:28 +0800
Subject: [PATCH 31/46] Exclude Emergency Dialer from recent app list.

Emergency Dialer should not show up in recent app list.

Bug: 112168722
Bug: 127585929
Test: Long press power key, tap Emergency, quit Emergency Dialer, and
check recent app list.
Change-Id: I2fe917d16ab23de6e82f9cad7b064e2837ec6490
Merged-In: I2fe917d16ab23de6e82f9cad7b064e2837ec6490
---
 .../android/systemui/globalactions/GlobalActionsDialog.java   | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/packages/SystemUI/src/com/android/systemui/globalactions/GlobalActionsDialog.java b/packages/SystemUI/src/com/android/systemui/globalactions/GlobalActionsDialog.java
index f72bc76995a..0fd27877247 100644
--- a/packages/SystemUI/src/com/android/systemui/globalactions/GlobalActionsDialog.java
+++ b/packages/SystemUI/src/com/android/systemui/globalactions/GlobalActionsDialog.java
@@ -523,7 +523,9 @@ class GlobalActionsDialog implements DialogInterface.OnDismissListener,
         public void onPress() {
             MetricsLogger.action(mContext, MetricsEvent.ACTION_EMERGENCY_DIALER_FROM_POWER_MENU);
             Intent intent = new Intent(EmergencyDialerConstants.ACTION_DIAL);
-            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP);
+            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK
+                    | Intent.FLAG_ACTIVITY_EXCLUDE_FROM_RECENTS
+                    | Intent.FLAG_ACTIVITY_CLEAR_TOP);
             intent.putExtra(EmergencyDialerConstants.EXTRA_ENTRY_TYPE,
                     EmergencyDialerConstants.ENTRY_TYPE_POWER_MENU);
             mContext.startActivityAsUser(intent, UserHandle.CURRENT);
-- 
2.17.1

